omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,196 @@
+
+		if (newConfiguration.getEnableVisits()) {
+			updateVisitEncounterHandlerValue(administrationService,
+					newConfiguration.getEncounterVisitsAssignmentHandler());
+		}
+		updateGetAutoCloseVisitsTaskStartedValue(schedulerService, newConfiguration.getStartAutoCloseVisitsTask());
+		updateVisitTypesToAutoCloseValue(administrationService, visitService, newConfiguration.getVisitTypesToAutoClose());
+	}
+
+	private Boolean getEnableVisitsValue(AdministrationService administrationService) {
+		String enableVisits = administrationService
+				.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, Boolean.FALSE.toString());
+		return Boolean.parseBoolean(enableVisits);
+	}
+
+	private void updateEnableVisitsValue(AdministrationService administrationService, Boolean value) {
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS,
+				Boolean.toString(value));
+		administrationService.saveGlobalProperty(enableVisitsGp);
+	}
+
+	private String getVisitEncounterHandlerValue(AdministrationService administrationService,
+			EncounterService encounterService) {
+		String visitEncounterHandler = administrationService.getGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER);
+		for (EncounterVisitHandler visitHandler : getEncounterVisitHandlers(encounterService)) {
+			if (visitHandler.getClass().getName().equals(visitEncounterHandler)) {
+				return visitHandler.getClass().getName();
+			}
+		}
+		return null;
+	}
+
+	private Collection<EncounterVisitHandler> getEncounterVisitHandlers(EncounterService encounterService) {
+		return encounterService.getEncounterVisitHandlers();
+	}
+
+	private void updateVisitEncounterHandlerValue(AdministrationService administrationService, String value) {
+		GlobalProperty visitEncounterHandlerGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER, value);
+		administrationService.saveGlobalProperty(visitEncounterHandlerGp);
+	}
+
+	private Boolean getAutoCloseVisitsTaskStartedValue(SchedulerService schedulerService) {
+		TaskDefinition autoCloseVisitsTaskStarted = schedulerService
+				.getTaskByName(OpenmrsConstants.AUTO_CLOSE_VISITS_TASK_NAME);
+
+		if (autoCloseVisitsTaskStarted != null) {
+			return autoCloseVisitsTaskStarted.getStarted();
+		}
+
+		return false;
+	}
+
+	private void updateGetAutoCloseVisitsTaskStartedValue(SchedulerService schedulerService,
+			Boolean autoCloseVisitsTaskStarted) {
+		TaskDefinition closeVisitsTask = schedulerService.getTaskByName(OpenmrsConstants.AUTO_CLOSE_VISITS_TASK_NAME);
+		if (closeVisitsTask != null) {
+			try {
+				if (autoCloseVisitsTaskStarted && !closeVisitsTask.getStarted()) {
+					schedulerService.scheduleTask(closeVisitsTask);
+				} else if (!autoCloseVisitsTaskStarted && closeVisitsTask.getStarted()) {
+					schedulerService.shutdownTask(closeVisitsTask);
+				}
+			}
+			catch (SchedulerException e) {
```

Comment:
Instead of catching this, how about just letting it bubble through?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/d976fea0bb3f182a33e04a7f4c77186d5658402f